libs/crypto/src/lib.rs

```diff
@@ -399,6 +400,30 @@ enum Kdf {
     },
 }
 
+#[derive(Debug, Clone, PartialEq)]
+pub enum PasswordError {
+    Base64Decoding,
+    InvalidFormat,
+    InvalidKeyLength,
+    InvalidLength,
+    InvalidSaltLength,
+    NoMatchingAlgorithm(String),
```

Comment:
I think didn't I mention we had a way to report this from the higher levels anyway, so we don't need the `(String)` here?

Reply:
Different PR but why not have the feed back and save having to parse it in a different part of the code?

Reply:
We already have the handler in the password import code though, it's just a string so all we have to return is a few characters. It also isn't a complete option, because on 819 you have `_ => Err(PasswordError::NoMatchingDecoder)` , which the current code *does* handle from the higher level. 

So if you really want to have the string here, it should have the same logic from https://github.com/kanidm/kanidm/blob/master/server/lib/src/plugins/cred_import.rs#L65 so that anytime we would have noMatchingDecoder, we would actually have `NoMatchingAlgorithm()` with the hint about what the algorithm is.

Reply:
So if it's not `{}`-wrapped (which the code handles) we should just return the first 4-5 chars of the input regardless?

Reply:
The intention with the difference was "we have no idea what this is" compared to "we don't support that algo currently" - I updated the names to `UnsupportedAlgorithm(String)` and `NoDecoderFound` to make it a little less confusing.

Reply:
I'm down with pink sheds as long as we know it's pink 😄